src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts

```diff
@@ -732,4 +732,28 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
     this.sessionsTableRowModels = [];
     this.recycleBinFeedbackSessionRowModels = [];
   }
+
+  triggerModelChange(data: SessionEditFormModel): void {
+    const { submissionStartDate, submissionEndDate, submissionStartTime, submissionEndTime } = data;
+
+    const startDate = new Date(submissionStartDate.year, submissionStartDate.month, submissionStartDate.day);
+    const endDate = new Date(submissionEndDate.year, submissionEndDate.month, submissionEndDate.day);
+
+    if (startDate > endDate) {
+      this.sessionEditFormModel = {
+        ...data,
+        submissionEndDate: data.submissionStartDate,
+        submissionEndTime,
+      };
+    } else if (startDate.toISOString() === endDate.toISOString() && submissionStartTime.hour > submissionEndTime.hour) {
+      this.sessionEditFormModel = {
+        ...data,
+        submissionEndDate: data.submissionStartDate,
+        submissionEndTime: {
+          ...submissionStartTime,
+          hour: submissionStartTime.hour,
+        },
+      };
+    }
```

Comment:
seems like the logic here is the same as in `src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts`, would it be possible to abstract this logic out?

Reply:
Hey @cedricongjh, thanks for the review! May I clarify if you meant to abstract this method to the base class? The `sessionEditFormModel` property is not on the base class, so this makes it a little tricky. Is there any pattern you have in mind?

Reply:
Hi @Zxun2, quickly jumping in for @cedricongjh here since I had the same thought (: The `InstructorSessionEditPageComponent` extends `InstructorSessionBasePageComponent` and the `InstructorSessionsPageComponent` extends `InstructorSessionModalPageComponent` which also extends `InstructorSessionBasePageComponent`. Hence, we can just place this method in the base class, so that both components inherit the method. This way, we will avoid the code duplication.

Worth noting that the `InstructorSessionBasePageComponent` already has several methods that are used in other child components, so we are just following the same design principle. Hope this helps!

Reply:
Thanks for the clarification, and apologies for the late reply! I have shifted `sessionEditFormModel` and the `triggerModelChange` method to the base class.